code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -176,6 +175,13 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 	diag_hud_set_borgcell()
 	scanner = new(src)
 	scanner.Grant(src)
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, /mob/living/silicon/robot/.proc/create_trail)
+
+/mob/living/silicon/robot/proc/create_trail(datum/source, atom/OldLoc, Dir, forced)
+	if(ionpulse_on)
+		var/turf/T = get_turf(OldLoc)
+		if(!has_gravity(T))
+			new /obj/effect/particle_effect/ion_trails(T, Dir)
```

Comment:
```suggestion
/mob/living/silicon/robot/proc/create_trail(datum/source, atom/oldloc, dir, forced)
	if(ionpulse_on)
		var/turf/T = get_turf(oldloc)
		if(!has_gravity(T))
			new /obj/effect/particle_effect/ion_trails(T, dir)
```

Reply:
I will do this for `oldloc`, but `dir` is keeping a different name to aid diffrentiation